Server/src/models/VariableModel.ts

```diff
@@ -0,0 +1,24 @@
+import { Datapoints } from "./entities/Datapoints";
+import { IVariableNameModel } from "../../../Client/src/Models/IVariableNameModel"
+import { getConnection, getManager } from "typeorm";
+
+
+export class VariableModel {
+  /**
+  * This method will return all existing variables or datapoints in the format of IVariableModel
+  */
+  async getAllVariables(): Promise<IVariableNameModel[]> {
+    let variables = await getManager()
+      .createQueryBuilder(Datapoints, 'Datapoints')
+      .select('name')
+      .distinct(true)
+      .getRawMany();
+
+    let variableModel = variables.map(variable => {
+      return {
+        name: variable.name
+      }
+    })
+    return variableModel;
```

Comment:
I dont think this return hits. You might just be returning the first name here no?

Reply:
No it goes through all the variables fetched from DB, and maps them into an object that contains only a name, which is what will be returned to the frontend